test/ProcessTest.php

```diff
@@ -59,6 +65,186 @@ public function testCommandCanRun() {
+            $this->assertInternalType('int', $process->getPid());
+        });
+    }
+
+    public function testGetWorkingDirectoryIsDefault() {
+        Loop::run(function () {
+            $process = new Process(self::CMD_PROCESS);
+            $this->assertSame(getcwd(), $process->getWorkingDirectory());
+        });
+    }
+
+    public function testGetWorkingDirectoryIsCustomized() {
+        Loop::run(function () {
+            $process = new Process(self::CMD_PROCESS, __DIR__);
+            $this->assertSame(__DIR__, $process->getWorkingDirectory());
+        });
+    }
+
+    public function testGetEnv() {
+        Loop::run(function () {
+            $process = new Process(self::CMD_PROCESS, null, []);
+            $this->assertCount(0, $process->getEnv());
+        });
+    }
+
+    public function testGetStdinIsCustomized() {
+        Loop::run(function () {
+            $process = new Process(self::CMD_PROCESS, null, [], [
+                new ResourceInputStream(fopen(__DIR__.'/../stream', 'r')),
+            ]);
+            $process->start();
+            $promise = $process->join();
+            $completed = false;
+            $promise->onResolve(function () use (&$completed) { $completed = true; });
+            $this->assertInstanceOf(ResourceOutputStream::class, $process->getStdin());
+        });
+    }
+
+    public function testGetStdoutIsCustomized() {
+        Loop::run(function () {
+            $process = new Process(self::CMD_PROCESS, null, [], [
+                new ResourceOutputStream(fopen(__DIR__.'/../stream', 'w')),
+            ]);
+            $process->start();
+            $promise = $process->join();
+            $completed = false;
+            $promise->onResolve(function () use (&$completed) { $completed = true; });
+            $this->assertInstanceOf(ResourceInputStream::class, $process->getStdout());
+        });
+    }
+
+    public function testGetStderrIsCustomized() {
+        Loop::run(function () {
+            $process = new Process(self::CMD_PROCESS, null, [], [
+                new ResourceOutputStream(fopen(__DIR__.'/../stream', 'w')),
+            ]);
+            $process->start();
+            $promise = $process->join();
+            $completed = false;
+            $promise->onResolve(function () use (&$completed) { $completed = true; });
+            $this->assertInstanceOf(ResourceInputStream::class, $process->getStderr());
+        });
+    }
+
+    public function testProcessEnvIsValid() {
+        $process = new Process(self::CMD_PROCESS, null, [
+            'env_value'
+        ]);
+        $process->start();
+        $promise = $process->join();
+        $completed = false;
+        $promise->onResolve(function () use (&$completed) { $completed = true; });
+        $this->assertSame('env_value', $process->getEnv()[0]);
+    }
+
+    public function testProcessDeferredIsFailed() {
+        $process = new Process(new \Exception());
+        $this->assertNull($process->start());
```

Comment:
This doesn't seem to test what the test name says.

Reply:
Ok....This is my testing method and I forget to remove it.